pyqtgraph/graphicsItems/InfiniteLine.py

```diff
@@ -37,79 +78,125 @@ def __init__(self, pos=None, angle=90, pen=None, movable=False, bounds=None):
                         for :func:`mkPen <pyqtgraph.mkPen>`. Default pen is transparent
                         yellow.
         movable         If True, the line can be dragged to a new position by the user.
+        hoverPen        Pen to use when drawing line when hovering over it. Can be any
+                        arguments that are valid for :func:`mkPen <pyqtgraph.mkPen>`.
+                        Default pen is red.
         bounds          Optional [min, max] bounding values. Bounds are only valid if the
                         line is vertical or horizontal.
+        label           if True, a label is displayed next to the line to indicate its
+                        location in data coordinates
+        textColor       color of the label. Can be any argument fn.mkColor can understand.
+        textFill        A brush to use when filling within the border of the text.
+        textLocation    A float [0-1] that defines the location of the text.
+        textShift       A float [0-1] that defines when the text shifts from one side to
+                        another.
+        textFormat      Any new python 3 str.format() format.
+        unit            If not None, corresponds to the unit to show next to the label
+        name            If not None, corresponds to the name of the object
         =============== ==================================================================
         """
-        
-        GraphicsObject.__init__(self)
-        
+
+        UIGraphicsItem.__init__(self)
+
         if bounds is None:              ## allowed value boundaries for orthogonal lines
             self.maxRange = [None, None]
         else:
             self.maxRange = bounds
         self.moving = False
-        self.setMovable(movable)
         self.mouseHovering = False
+
+        self.angle = ((angle+45) % 180) - 45
+        if textColor is None:
+            textColor = (200, 200, 200)
+        self.textColor = textColor
+        self.location = textLocation
+        self.shift = textShift
+        self.label = label
+        self.format = textFormat
+        self.unit = unit
+        self._name = name
```

Comment:
The purpose is to define a name, that can set by the user. I added this because I'm writing a list manager that will allow the user to interact with the content of a plotwidget (and the name of the item is based on this property). I can remove it if you prefer.